packages/lib/gpcircuits/scripts/util.ts

```diff
@@ -15,3 +26,149 @@ export async function clearDir(directory: string): Promise<void> {
  * to avoid OOM issues.
  */
 export const MAX_PARALLEL_PROMISES = 4;
+
```

Comment:
This is the core implementation, not really a utility.  Maybe they belong in a file called "common" or "impl" rather than "util".

Reply:
Note that I'm assuming the actual generation logic is unchanged other than the parameterization, so I'm not reviewing in detail.